velox/exec/RowContainer.cpp

```diff
@@ -560,6 +561,16 @@ void RowContainer::extractProbedFlags(
   }
 }
 
+int64_t RowContainer::estimateRowSize() const {
+  if (numRows_ == 0) {
+    return -1;
+  }
+  int64_t freeBytes = rows_.availableInRun() + fixedRowSize_ * numFreeRows_;
+  int64_t usedSize = rows_.allocatedBytes() - freeBytes +
+      stringAllocator_.retainedSize() - stringAllocator_.freeSpace();
+  return usedSize / numRows_;
```

Comment:
Can this produce a zero if usedSize < numRows_?

Is it possible for usedSize to be negative? Should we add VELOX_CHECK just in case?

Reply:
It makes sense. VELOX_CHECK is added. I originally thought leaving the caller to handle the edge cases.